src/IO/WriteBufferFromS3.h

```diff
@@ -88,9 +88,8 @@ class WriteBufferFromS3 final : public BufferWithOwnMemory<WriteBuffer>
 
     const String bucket;
     const String key;
-    const S3Settings::RequestSettings::PartUploadSettings settings;
-    const bool check_objects_after_upload = false;
-    const size_t max_unexpected_write_error_retries = 4;
+    const S3Settings::RequestSettings request_settings;
```

Comment:
what is the difference between them? Why we store them as separate fields?

Reply:
It's not a separate field, it's just const reference to a part of `request_settings` to make the implementation shorter. With this const reference I don't have to call `request_settings.getUploadSettings()` multiple times because I call it once in the constructor.

Reply:
I thought about it a bit more. Yes, probably we can try to find out which method should be used by checking the response of `HeadObject` - if it returns `400 Bad Request` then we can try `GetObjectAttributes`.